src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -1199,13 +1254,18 @@ void handle_task_submitted(LocalSchedulerState *state,
    * resource is currently unavailable, then consider queueing task locally and
    * recheck dynamic next time. */
 
-  /* If this task's constraints are satisfied, dependencies are available
-   * locally, and there is an available worker, then enqueue the task in the
-   * dispatch queue and trigger task dispatch. Otherwise, pass the task along to
-   * the global scheduler if there is one. */
+  // If this task's constraints are satisfied, dependencies are available
+  // locally, and there is an available worker, then enqueue the task in the
+  // dispatch queue and trigger task dispatch. Otherwise, pass the task along to
+  // the global scheduler if there is one. Note that if there are no CPUs on
+  // this machine, then we automatically give the task to the global scheduler.
+  // This is here to to prevent actor creation tasks which require 0 CPUs from
+  // being scheduled here because subsequent actor methods may require more
+  // CPUs.
   if (resource_constraints_satisfied(state, spec) &&
       (algorithm_state->available_workers.size() > 0) &&
-      can_run(algorithm_state, execution_spec)) {
+      can_run(algorithm_state, execution_spec) &&
+      state->static_resources["CPU"] != 0) {
```

Comment:
Might make more sense to move this logic into `resource_constraints_satisfied`. Then we can also add logic to make sure that it only happens for actor creation tasks.

Reply:
fixed